code/game/machinery/suit_storage_unit.dm

```diff
@@ -238,8 +284,30 @@
 	if(!(flags_1 & NODECONSTRUCT_1))
 		open_machine()
 		dump_inventory_contents()
+		if(card_reader_installed)
+			new /obj/item/stock_parts/card_reader(loc)
 	return ..()
 
+/obj/machinery/suit_storage_unit/proc/access_check(mob/living/user)
+	if(!isnull(id_card))
+		var/obj/item/card/id/id = id_card.resolve()
+		if(QDELETED(id))
+			name = initial(name)
+			desc = initial(desc)
+			id_card = null
+			req_access = list()
+			req_one_access = null
+			set_access(list())
+			return TRUE
+		if(id_card && (user.get_idcard() != id))
+			balloon_alert(user, "not your unit!")
+			return FALSE
+	else if(!allowed(user))
+		balloon_alert(user, "access denied!")
+		return FALSE
+
+	return TRUE
```

Comment:
```suggestion
	var/obj/item/card/id/id = id_card?.resolve()

	if(!id) // reset to defaults
		name = initial(name)
		desc = initial(desc)
		id_card = null
		req_access = list()
		req_one_access = null
		set_access(list())
		return TRUE

	if(user.get_idcard() != id)
		balloon_alert(user, "not your unit!")
		return FALSE

	if(!allowed(user))
		balloon_alert(user, "access denied!")
		return FALSE

	return TRUE
```

Reply:
@LemonInTheDark can you double check me on this since when I code dived to look at weakref patterns there were some inconsistencies between using `isnull(weakref_obj)` vs `QDELETED(weakref_obj)` vs `!weakref_obj`

Reply:
QDELETED() checks both if an object is null and is deleted so its the best option?

Reply:
resolve() won't ever return a qdeleted object, we assert that if something is qdeleted at any stage it doesn't "exist" anymore
![image](https://user-images.githubusercontent.com/58055496/235336182-a334b1c2-96bb-47e3-bc6b-d1cb3d3bbca1.png)

The instinct to check it is good, but in this case a null check (however you prefer moth keeps making me do isnull) is fine.

Reply:
I'll do that then